specs/phase1/beacon-chain.md

```diff
@@ -166,246 +91,74 @@ class AttestationData(Container):
-    exposed_derived_secrets: Vector[List[ValidatorIndex, MAX_EARLY_DERIVED_SECRET_REVEALS * SLOTS_PER_EPOCH],
-                                    EARLY_DERIVED_SECRET_PENALTY_MAX_FUTURE_EPOCHS]
-    custody_chunk_challenge_records: List[CustodyChunkChallengeRecord, MAX_CUSTODY_CHUNK_CHALLENGE_RECORDS]
-    custody_chunk_challenge_index: uint64
 ```
 
 ## New containers
 
 The following containers are new in Phase 1.
 
-### `ShardBlock`
+### `DataCommitment`
 
 ```python
-class ShardBlock(Container):
-    shard_parent_root: Root
-    beacon_parent_root: Root
-    slot: Slot
-    shard: Shard
-    proposer_index: ValidatorIndex
-    body: ByteList[MAX_SHARD_BLOCK_SIZE]
+class DataCommitment(Container):
+    # Kate commitment to the data
+    point: BLSCommitment
+    # Length of the data in samples
+    length: length
 ```
 
-### `SignedShardBlock`
+### `ShardHeader`
 
 ```python
-class SignedShardBlock(Container):
-    message: ShardBlock
-    signature: BLSSignature
-```
-
-### `ShardBlockHeader`
-
-```python
-class ShardBlockHeader(Container):
-    shard_parent_root: Root
-    beacon_parent_root: Root
+class ShardHeader(Container):
+    # Slot and shard that this header is intended for
     slot: Slot
     shard: Shard
-    proposer_index: ValidatorIndex
-    body_root: Root
-```
-
-### `ShardState`
-
-```python
-class ShardState(Container):
-    slot: Slot
-    gasprice: Gwei
-    latest_block_root: Root
+    # The actual data commitment
+    commitment: DataCommitment
+    # Proof of the length (more precisely, proof that values at
+    # positions >= the length all equal zero)
+    length_proof: BLSCommitment
 ```
 
-### `ShardTransition`
+### `PendingShardHeader`
 
 ```python
-class ShardTransition(Container):
-    # Starting from slot
-    start_slot: Slot
-    # Shard block lengths
-    shard_block_lengths: List[uint64, MAX_SHARD_BLOCKS_PER_ATTESTATION]
-    # Shard data roots
-    # The root is of ByteList[MAX_SHARD_BLOCK_SIZE]
-    shard_data_roots: List[Bytes32, MAX_SHARD_BLOCKS_PER_ATTESTATION]
-    # Intermediate shard states
-    shard_states: List[ShardState, MAX_SHARD_BLOCKS_PER_ATTESTATION]
-    # Proposer signature aggregate
-    proposer_signature_aggregate: BLSSignature
-```
-
-### `CompactCommittee`
-
-```python
-class CompactCommittee(Container):
-    pubkeys: List[BLSPubkey, MAX_VALIDATORS_PER_COMMITTEE]
-    compact_validators: List[uint64, MAX_VALIDATORS_PER_COMMITTEE]
+class PendingShardHeader(Container):
+    # Slot and shard that this header is intended for
+    slot: uint64
+    shard: Shard
+    # Kate commitment to the data
+    commitment: BLSCommitment
+    # hash_tree_root of the ShardHeader (stored so that attestations
+    # can be checked against it)
+    root: Hash
+    # Length of the data in samples
+    length: uint64
+    # Who voted for the header
+    votes: Bitlist[MAX_COMMITTEE_SIZE]
+    # Has this header been confirmed?
+    confirmed: bool
```

Comment:
Missing length proof?

Reply:
The PendingShardHeaders don't need a length proof, because the proof is already verified when the shard header gets accepted.